### PR TITLE
Add LogQL formatter

### PR DESCRIPTION
**Pull request by itsgareth**
## how to test this pr:

**in grafana/lezer-logql:**
1. checkout to this branch
2. build lezer from this branch

**in grafana/grafana**
1. checkout to the branch `gareth/prettify-query`
2. make sure you have enabled the feature flag `lokiFormatQuery`
3. add the following to package.json
```json
"resolutions": {
  "@grafana/lezer-logql": "portal:/path/to/your/repo/lezer-logql/dist"
}
```

! grafana should now be using this build of lezer, try formatting some queries.

### CODE REVIEW

**Comment by matyax** [comment]
Before merging this I would:
- Add and setup Jest (we can do this together).
- Add the formatter tests.

File: generate-types.sh [hunk 1]
```diff
@@ -28,4 +28,6 @@ import { LRParser } from '@lezer/lr'
 
 export const parser: LRParser
 $(sed -E 's/ = [0-9]+/: number/' src/parser.terms.js)
+
+export const formatLokiQuery: (query: string) => string
```
Comment (matyax) [inline generate-types.sh:32]: Do you know what this file does?

File: test/format.test.js [hunk 2]
```diff
@@ -0,0 +1,427 @@
...
+      );
+    });
+
+    it('handles label filters', () => {
+      expect(formatLokiQuery(`{label=""} | label=""`)).toBe(`{label=""}\n  | label=""`);
+      expect(formatLokiQuery(`{label=""} | label=ip("")`)).toBe(`{label=""}\n  | label=ip("")`);
+      expect(formatLokiQuery(`{label=""} | label=10s`)).toBe(`{label=""}\n  | label=10s`);
+      expect(formatLokiQuery(`{label=""} | label=1GB`)).toBe(`{label=""}\n  | label=1GB`);
+      expect(formatLokiQuery(`{label=""} | label=42`)).toBe(`{label=""}\n  | label=42`);
+      // expect(formatLokiQuery(`{label=""} | labelA="A" and labelB="B"`)).toBe(
```
Comment (matyax) [inline test/format.test.js:94]: What's going on here?

**Review by matyax — COMMENTED** [review]
Amazing! The implementation looks really really clean! 👏 👏 👏 👏 

Is there anything pending?

File: generate-types.sh (hunk 1 above)
Comment (gwdawson) [reply generate-types.sh:32]: yeah this generates the ts types that are exported with the package,
if we don't add to this, when importing `formatLokiQuery` into grafana we get the following error
`Module '"@grafana/lezer-logql"' has no exported member 'formatLokiQuery' .ts(2305)`

Comment (matyax) [reply generate-types.sh:32]: 👌

File: test/format.test.js (hunk 2 above)
Comment (gwdawson) [reply test/format.test.js:94]: these tests are commented out because they are failing.
these fixes are on my todo.

**Review by matyax — APPROVED** [review]
Let's gooooo! 🚀 ☄️

**Pull request merged**